File: app/src/main/java/org/scp/gymlog/room/entities/MuscleEntity.java
package org.scp.gymlog.room.entities;

import androidx.room.Embedded;
import androidx.room.Entity;
import androidx.room.Junction;
import androidx.room.PrimaryKey;
import androidx.room.Relation;

import java.util.List;

@Entity(tableName = "muscle")
public class MuscleEntity {
    @PrimaryKey
    public int muscleId;

    public static class WithExercises {
        @Embedded
        public MuscleEntity muscleGroup;
        @Relation(
                parentColumn = "muscleId",
                entityColumn = "exerciseId",
                associateBy = @Junction(ExerciseMuscleCrossRef.class)
        )
        public List<ExerciseEntity> exercises;
    }
}

File: app/src/main/java/org/scp/gymlog/room/entities/MuscleEntity.kt
package org.scp.gymlog.room.entities

import androidx.room.Embedded
import androidx.room.Entity
import androidx.room.Junction
import androidx.room.PrimaryKey
import androidx.room.Relation

@Entity(tableName = "muscle")
class MuscleEntity {
    @PrimaryKey
    var muscleId = 0

    class WithExercises {
        @Embedded
        lateinit var muscle: MuscleEntity

        @Relation(
            parentColumn = "muscleId",
            entityColumn = "exerciseId",
            associateBy = Junction(ExerciseMuscleCrossRef::class))
        var exercises: List<ExerciseEntity> = ArrayList()
    }
}